core/src/main/java/org/apache/accumulo/core/util/compaction/CompactionJobPrioritizer.java

```diff
@@ -19,42 +19,120 @@
 package org.apache.accumulo.core.util.compaction;
 
 import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
 
+import org.apache.accumulo.core.clientImpl.Namespace;
+import org.apache.accumulo.core.data.NamespaceId;
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.core.metadata.MetadataTable;
+import org.apache.accumulo.core.metadata.RootTable;
 import org.apache.accumulo.core.spi.compaction.CompactionJob;
 import org.apache.accumulo.core.spi.compaction.CompactionKind;
+import org.apache.accumulo.core.util.Pair;
+
+import com.google.common.base.Preconditions;
 
 public class CompactionJobPrioritizer {
 
   public static final Comparator<CompactionJob> JOB_COMPARATOR =
       Comparator.comparingInt(CompactionJob::getPriority)
           .thenComparingInt(job -> job.getFiles().size()).reversed();
 
-  public static short createPriority(CompactionKind kind, int totalFiles, int compactingFiles) {
+  private static final Map<Pair<TableId,CompactionKind>,Pair<Short,Short>> SYSTEM_TABLE_RANGES =
+      new HashMap<>();
+  private static final Map<Pair<NamespaceId,CompactionKind>,
+      Pair<Short,Short>> ACCUMULO_NAMESPACE_RANGES = new HashMap<>();
+
+  // Create ranges of possible priority values where each range has
+  // 2000 possible values. Give higher priority to tables where
+  // they have more files than allowed, user compactions over system
+  // compactions, root table over metadata table, metadata table over
+  // other system tables and user tables.
+  private static final Short TABLE_OVER_SIZE_MAX = Short.MAX_VALUE;
```

Comment:
Not sure these compactions should be given higher priority than root and metadata compactions.  Maybe this range could be moved between SYSTEM_NS_SYSTEM_MAX and USER_TABLE_USER_MAX.  Then the TABLE_OVER_SIZE range would only be used non accumulo namespaces.